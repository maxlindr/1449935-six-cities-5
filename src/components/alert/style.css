.alert {
  display: flex;
  align-items: center;
  justify-content: center;

  background-color: rgba(0, 0, 0, 0.4);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0
}

.alert__window {
  display: flex;
  flex-direction: column;
  flex-grow: 1;

  align-items: center;

  margin: 10px;
  padding: 30px;

  max-width: 400px;
  min-height: 200px;

  background-color: white;
}

.alert__container {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-grow: 1;
  margin-bottom: 30px;
}

.alert__message {
  text-align: center;
  margin: 0;
  padding: 0;
}

.alert__button {
  padding: 16px 20px 13px;
  min-width: 120px;

  color: #fff;
  background-color: #4481c3;
  border-radius: 3px;
}
