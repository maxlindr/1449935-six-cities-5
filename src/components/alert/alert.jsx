import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {ActionCreator} from '../../store/action';
import {getAlertMessage} from '../../store/selectors';
import './style.css';

export const Alert = ({message, close}) => {

  return (
    <aside className="alert">
      <div className="alert__window">
        <div className="alert__container">
          <p className="alert__message">{message}</p>
        </div>
        <button className="button alert__button" type="button" onClick={close}>
          OK
        </button>
      </div>
    </aside>
  );
};

Alert.propTypes = {
  message: PropTypes.string,
  close: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  message: getAlertMessage(state)
});

const mapDispatchToProps = (dispatch) => ({
  close: () => dispatch(ActionCreator.closeAlert()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Alert);

