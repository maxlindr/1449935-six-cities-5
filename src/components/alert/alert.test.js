import React from 'react';
import renderer from 'react-test-renderer';
import {Alert} from './alert';

const MESSAGE = `Hello Jest!`;

test(`Alert должен корректно отображаться`, () => {
  const tree = renderer.create(
      <Alert message={MESSAGE} onClose={() => {}}/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
