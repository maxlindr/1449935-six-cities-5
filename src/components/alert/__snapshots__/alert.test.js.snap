// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert должен корректно отображаться 1`] = `
<aside
  className="alert"
>
  <div
    className="alert__window"
  >
    <div
      className="alert__container"
    >
      <p
        className="alert__message"
      >
        Hello Jest!
      </p>
    </div>
    <button
      className="button alert__button"
      onClick={[Function]}
      type="button"
    >
      OK
    </button>
  </div>
</aside>
`;
